### PR TITLE
Add select a node before migration

### PR DESCRIPTION
**Pull request by chrisho**
**Description:**
Describe the matching rules for selecting nodes before migration.

**Related issue:**
https://github.com/harvester/harvester/issues/3032

**Backend code PR:**
https://github.com/harvester/harvester/pull/3399

### CODE REVIEW

File: docs/vm/live-migration.md [hunk 1]
```diff
@@ -48,3 +48,9 @@ Live migration will be aborted if it exceeds the completion timeout of 800s per
 ### Progress Timeout
 
 Live migration will also be aborted when copying memory doesn't make any progress in 150s.
+
+## Select Migratable Nodes
+
+When users have configured Node scheduling rules for a VM (**[Node Scheduling Tab](https://docs.harvesterhci.io/v1.1/vm/create-windows-vm/#node-scheduling-tab)**), ensure that the migrated nodes meet the VM's runtime requirements. When selecting the Nodes list before migration, the Nodes will be searched based on the scheduling rules parameters of the VM.
```
Comment (LucasSaintarbor) [inline docs/vm/live-migration.md:54]:
```suggestion
When you have [node scheduling rules](https://docs.harvesterhci.io/v1.1/vm/create-windows-vm/#node-scheduling-tab) configured for a VM, you must ensure that the nodes you are migrating to meet the VM's runtime requirements. The list of nodes you get to search and select from will be generated based on:
  - VM scheduling rules.
  - Possibly node rules from the NetworkConfig.
```

File: docs/vm/live-migration.md [hunk 2]
```diff
@@ -48,3 +48,9 @@ Live migration will be aborted if it exceeds the completion timeout of 800s per
...
+
+These Nodes are filtered based on the VM scheduling rules, and it's worth noting that the VM may include Node rules from the NetworkConfig.
```
Comment (LucasSaintarbor) [inline docs/vm/live-migration.md:56]:
```suggestion
```

**Review by LucasSaintarbor — COMMENTED** [review]
Lmk if these changes are okay. Thanks!

File: docs/vm/live-migration.md [hunk 3]
```diff
@@ -48,3 +48,9 @@ Live migration will be aborted if it exceeds the completion timeout of 800s per
 ### Progress Timeout
 
 Live migration will also be aborted when copying memory doesn't make any progress in 150s.
+
+## Select Migratable Nodes
+
+When you have [node scheduling rules](https://docs.harvesterhci.io/v1.1/vm/create-windows-vm/#node-scheduling-tab) configured for a VM, you must ensure that the nodes you are migrating to meet the VM's runtime requirements. The list of nodes you get to search and select from will be generated based on:
+  - VM scheduling rules.
+  - Possibly node rules from the NetworkConfig.
```
Comment (LucasSaintarbor) [inline docs/vm/live-migration.md:56]:
```suggestion
  - Possibly node rules from the network configuration.
```

**Review by LucasSaintarbor — APPROVED** [review]
Small nit 👍

**Review by guangbochen — COMMENTED** [review]
perfer to move the doc to the section as below
image

**Review by LucasSaintarbor — APPROVED** [review]

File: docs/vm/live-migration.md [hunk 4]
```diff
@@ -30,6 +30,10 @@ Live migration means moving a virtual machine to a different host without downti
 
 ![](/img/v1.2/vm/migrate-action.png)
 
+When you have [node scheduling rules](https://docs.harvesterhci.io/v1.1/vm/create-windows-vm/#node-scheduling-tab) configured for a VM, you must ensure that the nodes you are migrating to meet the VM's runtime requirements. The list of nodes you get to search and select from will be generated based on:
```
Comment (LucasSaintarbor) [inline docs/vm/live-migration.md:33]:
```suggestion
When you have [node scheduling rules](https://docs.harvesterhci.io/v1.1/vm/create-windows-vm/#node-scheduling-tab) configured for a VM, you must ensure that the target nodes you are migrating to meet the VM's runtime requirements. The list of nodes you get to search and select from will be generated based on:
```

**Review by LucasSaintarbor — APPROVED** [review]
Small change to use "target nodes" terms from UI

**Review by guangbochen — APPROVED** [review]

**Pull request merged**